__tests__/app.test.js

```diff
@@ -303,4 +303,26 @@ describe('GET /api/users', ()=>{
             expect(body.msg).toBe("Route not found")
         })
     })
+})
+
+describe('GET /api/articles?topic=cats', ()=>{
+    test('should filter responses based on the topic added in the query', ()=>{
+        return request(app)
+        .get('/api/articles?topic=cats')
+        .expect(200)
+        .then(({body})=>{
+            expect(body.articles.length).toBe(1)
+        })
+    })
+})
+
+describe('GET /api/articles?topic=cats', () =>{
+    test('should return 404 with misspelt query ', () => {
+        return request(app)
+        .get('/api/articles?topic=miss_spelt')
+        .expect(404)
+        .then(({body})=>{
+            expect(body.msg).toBe('Topic not found')
+        })
+    });
```

Comment:
There is another really important test missing here which is when we are querying for a topic that does exist but one that has no associated articles. I believe the topic paper is a valid topic but one that has no articles, so in that case we should expect an empty array as a response.